### PR TITLE
Update .net7

### LINKED ISSUE #1223: Upgrade to .NET 7
##### **Overview of the feature request**

.NET 7 will be released soon and it would be nice to upgrade the project to use .Net v7

##### **Related issues or PR**

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by 123445666**
This Pull Request include:

- Update .Net 7

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

resolved #1223

### CODE REVIEW

**Comment by ivanmonteiro** [comment]
Thanks for the PR. I'll take a look at it soon

File: generators/client/templates/blazor/test/Project.Client.Test/AlertErrorTest.cs.ejs [hunk 1]
```diff
@@ -49,7 +49,7 @@ public class AlertErrorTest : TestContext
             .Replace(ServiceDescriptor.Transient<IComponentActivator, ComponentActivator>())
             .AddBootstrapProviders()
             .AddFontAwesomeIcons();
-
+        JSInterop.Mode = JSRuntimeMode.Loose;
```
Comment (ivanmonteiro) [inline generators/client/templates/blazor/test/Project.Client.Test/AlertErrorTest.cs.ejs:52]: Why is this needed?

File: generators/server/templates/dotnetcore/src/Project/Configuration/DatabaseStartup.cs.ejs [hunk 2]
```diff
@@ -61,7 +61,7 @@ public static class DatabaseConfiguration
                     connectionString = $"Server={databaseUri.Host};Port={databaseUri.Port};Database={db};User={userInfo[0]};Password={userInfo[1]};Pooling=true;MinimumPoolSize=0;MaximumPoolsize=10;";
                     break;
                 case "mssql":
-                    connectionString = $"Server={databaseUri.Host};Port={databaseUri.Port};Database={db};User={userInfo[0]};Password={userInfo[1]};Trusted_Connection=False;Pooling=true;";
+                    connectionString = $"Server={databaseUri.Host};Port={databaseUri.Port};Database={db};User={userInfo[0]};Password={userInfo[1]};Trusted_Connection=False;Encrypt=False;Pooling=true;";
```
Comment (nicolas63) [inline generators/server/templates/dotnetcore/src/Project/Configuration/DatabaseStartup.cs.ejs:64]: Why disable encryption ? It's by default now ?

**Review by nicolas63 — COMMENTED** [review]
Very nice ! I'm excited to test this =)

File: generators/client/templates/blazor/test/Project.Client.Test/AlertErrorTest.cs.ejs (hunk 1 above)
Comment (123445666) [reply generators/client/templates/blazor/test/Project.Client.Test/AlertErrorTest.cs.ejs:52]: Hi @ivanmonteiro  
Without this setting, the test is showing like this photo 
image

I tried to add "SetupModule("./_content/Blazorise/utilities.js?v=1.2.2.0")"
But it's so annoying with this error after
" Bunit.JSRuntimeUnhandledInvocationException : bUnit's JSInterop has not been configured to handle the call:

    InvokeVoidAsync("log", "%cThank you for using the free version of the Blazorise component library! We're happy to offer it to you for personal use. If you'd like to remove this message, consider purchasing a commercial license from https://blazorise.com/commercial. We appreciate your support!", System.String[])

Configure bUnit's JSInterop to handle the call with following:

    SetupVoid("log", "%cThank you for using the free version of the Blazorise component library! We're happy to offer it to you for personal use. If you'd like to remove this message, consider purchasing a commercial license from https://blazorise.com/commercial. We appreciate your support!", System.String[])
or the following, to match any arguments:
    SetupVoid("log", _ => true)

The setup methods are available on an instance of the BunitJSInterop or
BunitJSModuleInterop type. The standard BunitJSInterop is available
through the TestContext.JSInterop property, and a BunitJSModuleInterop
instance is returned from calling SetupModule on a BunitJSInterop instance.
"

image

Maybe we can find out another way after update or if you have any other idea for this.

File: generators/server/templates/dotnetcore/src/Project/Configuration/DatabaseStartup.cs.ejs (hunk 2 above)
Comment (123445666) [reply generators/server/templates/dotnetcore/src/Project/Configuration/DatabaseStartup.cs.ejs:64]: Yep, I worked with some project .Net 7 now and without this setting, we have to check "Trust server certificate" in SQL Server or it'll show the error of certificate when open the connection to SQL Server
image

Comment (123445666) [reply generators/server/templates/dotnetcore/src/Project/Configuration/DatabaseStartup.cs.ejs:64]: Or we can use the setting ";Encrypt=True;TrustServerCertificate=True" for the same way.

Comment (123445666) [reply generators/server/templates/dotnetcore/src/Project/Configuration/DatabaseStartup.cs.ejs:64]: @nicolas63 I've updated .NET 7 for jhipster-net . I can make a PR for it if you want

Comment (nicolas63) [reply generators/server/templates/dotnetcore/src/Project/Configuration/DatabaseStartup.cs.ejs:64]: Yes please

Comment (123445666) [reply generators/server/templates/dotnetcore/src/Project/Configuration/DatabaseStartup.cs.ejs]: Ah I did , there is https://github.com/jhipster/jhipster-net/pull/14 ^^!

**Comment by nicolas63** [comment]
LGTM, @ivanmonteiro if it's ok for you we can merge it

**Comment by nicolas63** [comment]
Can you update the version of JHipsterNet.Core to 1.0.9 and after that i can merge it

**Comment by 123445666** [comment]
Hi @nicolas63, I think jhipster.net v1.0.9 on nuget is for .net6, do you think we should increase version for it, don't you?
And in the jhipster-dotnetcore, I saw the library jhipster core doesn't have the version, I will rêchck about this.

**Comment by nicolas63** [comment]
> Hi @nicolas63, I think jhipster.net v1.0.9 on nuget is for .net6, do you think we should increase version for it, don't you?
> And in the jhipster-dotnetcore, I saw the library jhipster core doesn't have the version, I will rêchck about this.

Sorry it's my bad in the jhipster-net the update is manual here https://github.com/jhipster/jhipster-net/blob/main/.github/workflows/buildandpublish.yml#L25

Can you update to 1.0.10 ?

**Comment by 123445666** [comment]
> > Hi @nicolas63, I think jhipster.net v1.0.9 on nuget is for .net6, do you think we should increase version for it, don't you?
> > And in the jhipster-dotnetcore, I saw the library jhipster core doesn't have the version, I will rêchck about this.
> 
> Sorry it's my bad in the jhipster-net the update is manual here https://github.com/jhipster/jhipster-net/blob/main/.github/workflows/buildandpublish.yml#L25
> 
> Can you update to 1.0.10 ?

Done

**Comment by nicolas63** [comment]
everythings is good ! thanks @123445666
